lightning/src/ln/channelmanager.rs

```diff
@@ -10502,6 +10502,26 @@ where
 			"Dual-funded channels not supported".to_owned(),
 			 msg.channel_id.clone())), *counterparty_node_id);
 	}
+
+	fn handle_message_received(&self) {
+		let invoice_requests = self.pending_outbound_payments.get_invoice_request_awaiting_invoice();
+		if invoice_requests.is_empty() {
+			return;
+		}
+		let mut pending_offers_messages = self.pending_offers_messages.lock().unwrap();
+		for (payment_id, invoice_request) in invoice_requests {
+			let context = OffersContext::OutboundPayment { payment_id };
+			let reply_path = match self.create_blinded_path(context) {
+				Ok(path) => path,
+				Err(_) => return,
```

Comment:
We'll probably want to log here. This also means subsequent calls will not result in any retries since `get_invoice_request_awaiting_invoice` has already cleared the invoice requests. But maybe there isn't much we can do if the router fails.

Reply:
Introduce a log message in [fa5cf60](https://github.com/lightningdevkit/rust-lightning/commit/fa5cf6003f56940682f578688cd33c68838086e9)
Thanks for the suggestion!